geolib/models/dstability/internal.py

```diff
@@ -297,7 +297,7 @@ class PersistableStateLinePoint(DStabilityBaseModelStructure):
     IsProbabilistic: Optional[bool] = None
     Label: Optional[str] = ""
     Notes: Optional[str] = ""
-    X: Optional[float] = None
+    X: Optional[Union[float, str]] = "NaN"
```

Comment:
You can use: float('NaN') such that the X can no longer be of a string type.
Idem for other occurrences below

Reply:
I adhered to what was already implemented in e.g. PersistablePoint:
```
# waternet schema
class PersistablePoint(DStabilityBaseModelStructure):
    X: Optional[Union[float, str]] = "NaN"
    Z: Optional[Union[float, str]] = "NaN"
```
Not sure if it's worth the effort to do things differently here. Wdyt?

Reply:
@wfaustmann parameters with declaration `Optional[float] = float("NaN")` are serialized to `null` instead of `"NaN"`. So the declaration `Optional[Union[float, str]] = "NaN"` seems to be valid.